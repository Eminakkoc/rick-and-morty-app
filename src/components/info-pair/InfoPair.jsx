import React from 'react';
import PropTypes from 'prop-types';

const InfoPair = (props) => {
  const { label, value } = props;

  return (
    <div className="info-pair-container">
      <div className="info-pair-key">
        { `${label}: ` }
      </div>
      <div className="info-pair-value">
        { value }
      </div>
    </div>
  );
};

InfoPair.propTypes = {
  label: PropTypes.string,
  value: PropTypes.string,
};

InfoPair.defaultProps = {
  label: '*',
  value: '*',
};

export default InfoPair;
