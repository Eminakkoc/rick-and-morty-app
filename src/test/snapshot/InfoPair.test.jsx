// InfoPair.test.js
import React from 'react';
import InfoPair from '../../components/info-pair/InfoPair';
import renderer from 'react-test-renderer';

test('Info Pair Component Structure', () => {
  const component = renderer.create(
    <InfoPair />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
