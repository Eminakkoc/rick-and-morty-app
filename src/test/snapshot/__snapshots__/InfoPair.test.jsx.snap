// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info Pair Component Structure 1`] = `
<div
  className="info-pair-container"
>
  <div
    className="info-pair-key"
  >
    *: 
  </div>
  <div
    className="info-pair-value"
  >
    *
  </div>
</div>
`;
